scigym/envs/quantum_physics/quantum_computing/teleportation/README.md

```diff
@@ -22,17 +22,71 @@ where the input qubit is an arbitrary state Ψ.
 * Hadamard gate H on each qubit (3 actions)
 * CNOT-gate on the two qubits at location A (1 actions)
 * Z-measurements on each qubit (3 actions)
+
 In total: **10 actions**
 
-The Z-measurements are considered to be destructive, i.e. the qubit is removed and actions acting on that qubit will are no longer valid (and will not change the state).
+The Z-measurements are considered to be destructive, i.e. the qubit is removed and actions acting on that qubit are no longer valid (and will not change the state).
 If the agent can handle changing action sets, the currently available actions can be accessed via the attribute `env.available_actions` or the info dictionary returned by the `env.step(action)` method.
 
 *Observations*
+
 The observations perceived by the agent are a list of previous actions and measurement outcomes. Invalid actions (on qubits that have already been measured) are not added to this list.
 
+*Reward*
+
+If qubit B is in state Ψ, the protocol is considered successful and the agent receives a reward of 1.
+
+
+#### Known solution
+
+The quantum teleportation protocol in this formulation is performed by:
+
+1. Hadamard on qubit A'
+2. CNOT with qubit A' as the source and qubit A as the target
+3. Measure qubit A'
+4. Measure qubit A
 
-#### Known solutions
+Then, depending on the measurement outcome, a correction needs to be applied on qubit B. This is (equally likely):
+* Nothing
+* Pauli-Z correction: T^4
+* Pauli-X correction: H T^4 H
+* Pauli-Y correction: T^4 H T^4 H (up to global phase)
+
+So, depending on the measure outcome, the favored solution will either need
+4, 8, 10 or 14 actions.
+
+In order to have discovered a functioning protocol, the agent should find all those solutions, i.e. it should always be successful regardless of the measurement outcomes.
 
 #### Implementation details
 
+In order to make sure that the solution does not depend on the input state,
+we compare the Jamiolkowski fidelity of the effective channel generated by the
+gate sequence picked by the agent with the ideal channel. We do this by
+initializing the qubit A' in a state where it is entangled with an
+auxiliary qubit in a maximally entangled state Φ<sup>+</sup>.
+
+*Careful:* Because the environment will return `done` as `True` once no valid
+actions remain, this can lead to very short, but unrewarded trials.
+
 #### Discussion
```

Comment:
Do you not want to add your paper reference somewhere?

Reply:
That's probably a good idea.